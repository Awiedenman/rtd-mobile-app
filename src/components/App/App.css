.App {
  text-align: center;
  background: white;
  height: 100vh;
}

.overlay {
  height: 100%;
  width: 100%;
}
