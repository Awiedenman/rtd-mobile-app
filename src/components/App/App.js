import React from 'react';
import { Route, Switch } from 'react-router-dom';
import HomePage from '../Home/Home';
import Footer from '../Footer/Footer';
import Header from '../Header/Header';
import AccountPage from '../Account/Account';
import LandingPage from '../Landing/Landing';
import SignUpPage from '../../containers/SignUp/SignUp';
import SignInPage from '../../containers/SignIn/SignIn';
import PasswordForgetPage from '../PasswordForget/PasswordForget';
import withAuthentication from '../withAuthentication';
import ItineraryPage from '../../containers/ItineraryPage/ItineraryPage';
 
import * as routes from '../../constants/routes';

import './App.css';

export const App = () => {
  return (
    <div className="App">
      <div className="overlay">
        <Header />
        <Switch>
          <Route 
            exact path={routes.LANDING} 
            component={LandingPage} 
          />
          <Route 
            exact path={routes.SIGN_UP} 
            component={SignUpPage}
          />
          <Route 
            exact path={routes.SIGN_IN} 
            component={SignInPage} 
          />
          <Route 
            exact path={routes.PASSWORD_FORGET} 
            component={PasswordForgetPage} 
          />
          <Route 
            exact path={routes.HOME} 
            component={HomePage} 
          />
          <Route 
            exact path={routes.ACCOUNT} 
            component={AccountPage} 
          />
          <Route 
            exact path={routes.ITINERARY}
            component={ItineraryPage}
          />
        </Switch>
        <Footer />
      </div>
    </div>

  );
};


export default withAuthentication(App);
