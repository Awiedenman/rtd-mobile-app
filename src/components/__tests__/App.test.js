import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App/App';
import { shallow } from 'enzyme';


describe('App', () => {
  it('should matchSnapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});



