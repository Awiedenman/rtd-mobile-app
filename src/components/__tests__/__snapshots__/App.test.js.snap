// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should matchSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithAuthentication />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Unknown />,
      "value": null,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "unstable_read": [Function],
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Unknown />,
        "value": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "unstable_read": [Function],
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
